src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -1646,9 +1643,9 @@ public static object ConvertPSObjectToType(PSObject valueToConvert, Type resultT
         /// <returns></returns>
         public static T ConvertTo<T>(object valueToConvert)
         {
-            if (valueToConvert is T)
+            if (valueToConvert is T variable)
             {
-                return (T)valueToConvert;
+                return variable;
             }
             return (T)ConvertTo(valueToConvert, typeof(T), true, CultureInfo.InvariantCulture, null);
```

Comment:
Make sense to use `return ConvertTo( ... ) as T` as more fast?

Reply:
I think `ConvertTo() as T` is not faster. And `(T)ConvertTo()` ensures an exception to be thrown when `ConverTo()` doesn't really return a `T` object, which is the expected behavior.

Reply:
I think all exceptions occur in `ConvertTo(valueToConvert, typeof(T), true, CultureInfo.InvariantCulture, null);` and if the method returns a result then the result is always of type T. So formally we can use `as T`. I see there are performance comparisons of  `cast` and `as` operators - last is faster - but I don't know whether it is true for .Net Core 2.1.